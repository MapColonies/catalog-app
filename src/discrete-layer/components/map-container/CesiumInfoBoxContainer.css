.MCCesiumInfoBox {
    width: 100%;
    height: 100%;
    color: var(--mdc-theme-text-primary-on-dark);
    background-color: var(--mdc-theme-surface);
}