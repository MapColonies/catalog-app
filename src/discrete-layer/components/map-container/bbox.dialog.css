#bboxDialog .closeIcon {
  margin-left: auto;
}

body[dir='rtl'] #bboxDialog .closeIcon {
  margin-left: unset;
  margin-right: auto;
}

#bboxDialog .bboxForm {
  padding-top: 16px;
  font-size: 0.875rem;
}

#bboxDialog .bboxRow {
  display: flex;
  margin-bottom: 16px;
}

body[dir='rtl'] #bboxDialog .bboxRow {
  flex-direction: row-reverse;
}

#bboxDialog .bboxField {
  margin: 0 8px;
}

#bboxDialog .footer {
  display: flex;
  justify-content: flex-start;
  gap: 20px;
  height: 60px;
}

#bboxDialog .buttons {
  display: flex;
  gap: 20px;
  align-items: center;
  margin-left: auto;
}

body[dir='rtl'] #bboxDialog .buttons {
  margin-left: unset;
  margin-right: auto;
}

#bboxDialog .messages {
  display: flex;
  flex-direction: column;
}