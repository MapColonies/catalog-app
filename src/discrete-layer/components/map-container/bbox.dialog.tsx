import React, { useCallback, useState } from 'react';
import { FormattedMessage, useIntl, IntlShape } from 'react-intl';
import { useFormik } from 'formik';
import { isEmpty } from 'lodash';
import * as Yup from 'yup';
import * as turf from '@turf/helpers';
import distance from '@turf/distance/dist/js'; //TODO: make a consumption "REGULAR"
import {
  Dialog,
  DialogTitle,
  DialogContent,
  TextField,
  Button,
  IconButton,
} from '@map-colonies/react-core';
import { BboxCorner, Box, DrawType, IDrawingEvent } from '@map-colonies/react-components';
import CONFIG from '../../../common/config';
import { ValidationsError } from '../../../common/components/error/validations.error-presentor';
import { FieldLabelComponent } from '../../../common/components/form/field-label';
import { BBoxCorner, Corner } from '../bbox/bbox-corner-indicator';

import './bbox.dialog.css';

const NONE = 0;

interface BBoxCorners {
  topRightLat: number;
  topRightLon: number;
  bottomLeftLat: number;
  bottomLeftLon: number;
}

interface BBoxCornersError {
  latDistance?: string;
  lonDistance?: string;
  invalid?: string;
}

const validate = (values: BBoxCorners, intl: IntlShape): BBoxCornersError => {
  const errors: BBoxCornersError = {};

  try {
    turf.lineString([
      [values.bottomLeftLon, values.bottomLeftLat],
      [values.topRightLon, values.topRightLat],
    ]);

    const yDistance = distance(
      [values.bottomLeftLon, values.bottomLeftLat],
      [values.bottomLeftLon, values.topRightLat]
    );

    const xDistance = distance(
      [values.bottomLeftLon, values.bottomLeftLat],
      [values.topRightLon, values.bottomLeftLat]
    );

    if (yDistance > CONFIG.BOUNDARIES.MAX_Y_KM) {
      errors.latDistance = intl.formatMessage({
        id: 'custom-bbox.form-error.y-distance.text',
      });
    }

    if (xDistance > CONFIG.BOUNDARIES.MAX_X_KM) {
      errors.lonDistance = intl.formatMessage({
        id: 'custom-bbox.form-error.x-distance.text',
      });
    }
  } catch (err) {
    errors.invalid = intl.formatMessage({
      id: 'custom-bbox.form-error.invalid.text',
    });
  }

  return errors;
};

interface BBoxDialogProps {
  isOpen: boolean;
  onSetOpen: (open: boolean) => void;
  onPolygonUpdate: (polygon: IDrawingEvent) => void;
}

export const BBoxDialog: React.FC<BBoxDialogProps> = (props) => {
  const { isOpen, onSetOpen, onPolygonUpdate } = props;
  const intl = useIntl();
  const [corners] = useState<BBoxCorners>({
    topRightLat: 0,
    topRightLon: 0,
    bottomLeftLat: 0,
    bottomLeftLon: 0,
  });
  const yupSchema: Record<string, any> = {};
  Object.keys(corners).forEach(fieldName => {
    const fieldLabel = `custom-bbox.dialog-field.${fieldName}.label`;
    yupSchema[fieldName] = Yup
    .number()
    .typeError(
      intl.formatMessage(
        { id: 'validation-general.number' },
        { fieldName: `<strong>${intl.formatMessage({ id: fieldLabel })}</strong>` }
      )
    )
    .required(
      intl.formatMessage(
        { id: 'validation-general.required' },
        { fieldName: `<strong>${intl.formatMessage({ id: fieldLabel })}</strong>` }
      )
    );
  });

  const formik = useFormik({
    initialValues: corners,
    validationSchema: Yup.object({
      ...yupSchema
    }),
    onSubmit: (values) => {
      const errors = validate(values, intl);
      if (errors.latDistance === undefined && errors.lonDistance === undefined && errors.invalid === undefined) {
        try {
          onPolygonUpdate({
            primitive: undefined,
            type: DrawType.BOX,
            geojson: {
              type : 'FeatureCollection',
              features: [
                { 
                  type : 'Feature', 
                  properties : {  
                    type : BboxCorner.TOP_RIGHT,
                  }, 
                  geometry : { 
                    type : 'Point', 
                    coordinates : [ values.topRightLon, values.topRightLat ] 
                  }
                },
                { 
                  type : 'Feature', 
                  properties : {  
                    type : BboxCorner.BOTTOM_LEFT
                  }, 
                  geometry : { 
                    type : 'Point', 
                    coordinates : [ values.bottomLeftLon, values.bottomLeftLat ]  
                  }
                }
              ]
            }
          });
          closeDialog();
          setFormErrors({});
        } catch(e) {
          console.error(e);
        }
      } else {
        setFormErrors(errors);
      }
    },
  });

  const [formErrors, setFormErrors] = useState<BBoxCornersError>({});

  const closeDialog = useCallback(() => {
    onSetOpen(false);
  }, [onSetOpen]);

  const getValidationErrors = (errors: Record<string, any>): Record<string, string[]> => {
    const validationResults: Record<string, string[]> = {};
    Object.entries(errors).forEach(([key, value]) => {
      validationResults[key] = [ value as string ];
    });
    return validationResults;
  };

  return (
    <Box id="bboxDialog">
      <Dialog open={isOpen} preventOutsideDismiss={true}>
        <DialogTitle>
          <FormattedMessage id="custom-bbox.dialog.title" />
          <IconButton
            className="closeIcon mc-icon-Close"
            label="CLOSE"
            onClick={(): void => { closeDialog(); }}
          />
        </DialogTitle>
        <DialogContent>
          <form onSubmit={formik.handleSubmit} className="bboxForm" noValidate>
            <Box className="bboxRow">
              <Box className="bboxField">
                <FieldLabelComponent value='custom-bbox.dialog-field.topRightLat.label' isRequired={true} showTooltip={false}></FieldLabelComponent>
                <TextField
                  name="topRightLat"
                  type="number"
                  onChange={formik.handleChange}
                  value={formik.values.topRightLat}
                  required={true}
                />
              </Box>
              <Box className="bboxField">
                <FieldLabelComponent value='custom-bbox.dialog-field.topRightLon.label' isRequired={true} showTooltip={false}></FieldLabelComponent>
                <TextField
                  name="topRightLon"
                  type="number"
                  onChange={formik.handleChange}
                  value={formik.values.topRightLon}
                  required={true}
                />
              </Box>
              <BBoxCorner corner={Corner.TOP_RIGHT} className="bboxField"/>
            </Box>
            <Box className="bboxRow">
              <Box className="bboxField">
                <FieldLabelComponent value='custom-bbox.dialog-field.bottomLeftLat.label' isRequired={true} showTooltip={false}></FieldLabelComponent>
                <TextField
                  name="bottomLeftLat"
                  type="number"
                  onChange={formik.handleChange}
                  value={formik.values.bottomLeftLat}
                  required={true}
                />
              </Box>
              <Box className="bboxField">
                <FieldLabelComponent value='custom-bbox.dialog-field.bottomLeftLon.label' isRequired={true} showTooltip={false}></FieldLabelComponent>
                <TextField
                  name="bottomLeftLon"
                  type="number"
                  onChange={formik.handleChange}
                  value={formik.values.bottomLeftLon}
                  required={true}
                />
              </Box>
              <BBoxCorner corner={Corner.BOTTOM_LEFT} className="bboxField"/>
            </Box>
            <Box className="footer">
              <Box className="messages">
                {
                  !isEmpty(formik.errors) && 
                  <ValidationsError errors={getValidationErrors(formik.errors)}/>
                }
                {
                  isEmpty(formik.errors) &&
                  !isEmpty(formErrors) &&
                  <ValidationsError errors={getValidationErrors(formErrors)}/>
                }
              </Box>
              <Box className="buttons">
                <Button raised type="submit" disabled={Object.keys(formik.errors).length > NONE}>
                  <FormattedMessage id="general.ok-btn.text"/>
                </Button>
                <Button type="button" onClick={(): void => { closeDialog(); }}>
                  <FormattedMessage id="general.cancel-btn.text"/>
                </Button>
              </Box>
            </Box>
          </form>
        </DialogContent>
      </Dialog>
    </Box>
  );
};
