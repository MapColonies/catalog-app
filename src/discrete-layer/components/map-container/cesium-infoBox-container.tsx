import React from 'react';

export const CesiumInfoBoxContainer: React.FC<{ theme: Record<string, string> }> = (props) => {
    return <div style={{ backgroundColor: props.theme.surface, width: '100%' }} className='MCCesiumInfoBox'>
        { props.children }
    </div>
}