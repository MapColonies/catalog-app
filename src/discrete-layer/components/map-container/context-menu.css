.imageryContextMenuTheme {
  --context-menu-width: 260px;
  --order-width: 60px;
  --scrollbar-width: 16px;
}

.imageryContextMenu {
  background: var(--mdc-theme-surface);
  position: absolute;
  border-radius: 4px;
  padding: 12px;
  width: var(--context-menu-width);
}

.imageryContextMenuTitle {
  color: var(--mdc-theme-primary);
  font-weight: bold;
  display: inline-block;
  max-width: calc(var(--context-menu-width) - 52px);
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  direction: ltr;
}

.imageryContextMenuTitleInfo {
  color: var(--mdc-theme-primary);
  cursor: help;
}

.imageryMenu {
  width: 100%;
}

.imageryMenuIcon {
  vertical-align: sub;
  margin-right: 8px;
}

body[dir='rtl'] .imageryMenuIcon {
  margin-right: unset;
  margin-left: 8px;
}

.imageryContextMenuSubTitle {
  font-weight: bold;
  margin-top: 12px;
}

.imageryContextMenuSubTitleIcon {
  vertical-align: middle;
  color: var(--mdc-theme-primary);
  cursor: pointer;
}

.imageryContextMenuTableContainer {
  max-height: 100px;
  overflow: auto;
}

.imageryContextMenuTable {
  height: 100%;
}

.imageryContextMenuTableOrder {
  width: var(--order-width);
  text-align: center;
}

.imageryContextMenuTableField {
  max-width: calc(var(--context-menu-width) - var(--order-width) - var(--scrollbar-width) - 24px);
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  direction: ltr;
}

.imageryContextMenuTableSelectedRow {
  color: var(--mdc-theme-primary);
  font-weight: bold;
}

.imageryContextMenuEmpty {
  background: var(--mdc-theme-surface);
  position: absolute;
  border-radius: 4px;
}