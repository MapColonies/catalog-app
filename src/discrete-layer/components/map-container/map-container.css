.map {
  /* left: 20%;
  width: 80%;
  position: absolute; */
}

.filtersPosition {
  position: absolute;
  z-index: 1000;
  left: 24px;
  top: 12px;
}

body[dir='rtl'] .filtersPosition {
  left: unset;
  right: 24px;
}

.filtersContainer {
  display: flex;
  padding: 8px;
}

.filtersMargin {
  margin-left: 8px;
  display: flex;
  gap: 8px;
}

.drawerPosition{
  overflow: hidden; 
  position: absolute;
  top: 60px;
  left: 0px;
}

.drawerPosition .mdc-drawer{
  width: 100%;
} 
body[dir='rtl'] .drawerPosition {
  right: 0px;
  left: unset;
}
