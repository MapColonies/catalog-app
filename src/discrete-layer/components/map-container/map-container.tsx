import React, { useState } from 'react';
import { Feature, FeatureCollection, Point, Polygon } from 'geojson';
import { find } from 'lodash';
import { lineString } from '@turf/helpers';
import bbox from '@turf/bbox';
import bboxPolygon from '@turf/bbox-polygon';
import { 
  DrawType, 
  BboxCorner,
  CesiumMap, 
  CesiumDrawingsDataSource,
  CesiumColor,
  CesiumSceneMode,
  IDrawingEvent,
  IDrawing
} from '@map-colonies/react-components';
import { useTheme } from '@map-colonies/react-core';
import CONFIG from '../../../common/config';
import { PolygonSelectionUi } from './polygon-selection-ui';
import './map-container.css';
import { MOCK_DATA_IMAGERY_LAYERS_ISRAEL } from '../../../__mocks-data__/search-results.mock';
import { CesiumXYZLayer } from '@map-colonies/react-components/dist/cesium-map/layers/xyz.layer';

interface IDrawingObject {
  type: DrawType;
  handler: (drawing: IDrawingEvent) => void;
}

const getTimeStamp = (): string => new Date().getTime().toString();
const noDrawing: IDrawingObject = {
  type: DrawType.UNKNOWN,
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  handler: (drawing: IDrawingEvent) => {},
};
const DRAWING_MATERIAL_OPACITY = 0.5;
const DRAWING_MATERIAL_COLOR = CesiumColor.YELLOW.withAlpha(DRAWING_MATERIAL_OPACITY);
const DRAWING_OUTLINE_COLOR = CesiumColor.AQUA;

export interface MapContainerProps {
  handlePolygonSelected: (polygon: Polygon) => void;
  handlePolygonReset: () => void;
  handleOtherDrawers: () => void;
  mapActionsWidth: string;
  mapContent?: React.ReactNode;
  filters?: React.ReactNode[];
}

export const MapContainer: React.FC<MapContainerProps> = (props) => {
  const [isDrawing, setIsDrawing] = useState<boolean>(false);
  const [drawPrimitive, setDrawPrimitive] = useState<IDrawingObject>(noDrawing);
  const [drawEntities, setDrawEntities] = useState<IDrawing[]>([
    {
      coordinates: [],
      name: '',
      id: '',
      type: DrawType.UNKNOWN,
    },
  ]);
  const theme = useTheme();
  const [center] = useState<[number, number]>(CONFIG.MAP.CENTER as [number, number]);
  // const [showDefaulImagery] = useState<false | undefined>(CONFIG.ACTIVE_LAYER === 'OSM_LAYER'? undefined: false);
  
  const createDrawPrimitive = (type: DrawType): IDrawingObject => {
    return {
      type: type,
      handler: (drawing: IDrawingEvent): void => {
        const timeStamp = getTimeStamp();

        setIsDrawing(false);

        props.handlePolygonSelected((drawing.geojson as Feature).geometry as Polygon);

        setDrawEntities([
          {
            coordinates: drawing.primitive,
            name: `${type.toString()}_${timeStamp}`,
            id: timeStamp,
            type: drawing.type,
          },
        ]);
      },
    };
  };
  
  const setDrawType = (drawType: DrawType): void =>{
    setIsDrawing(true);
    setDrawPrimitive(createDrawPrimitive(drawType));
  }
 
  const onPolygonSelection = (polygon: IDrawingEvent): void => {
    const timeStamp = getTimeStamp();
    const bottomLeftPoint = find((polygon.geojson as FeatureCollection<Point>).features, (feat: Feature<Point>)=>{
      return feat.properties?.type === BboxCorner.BOTTOM_LEFT;
    }) as Feature<Point>;
    const rightTopPoint = find((polygon.geojson as FeatureCollection<Point>).features, (feat: Feature<Point>)=>{
      return feat.properties?.type === BboxCorner.TOP_RIGHT;
    }) as Feature<Point>;
    const line = lineString([
      [
        bottomLeftPoint.geometry.coordinates[0],
        bottomLeftPoint.geometry.coordinates[1]
      ],
      [
        rightTopPoint.geometry.coordinates[0],
        rightTopPoint.geometry.coordinates[1],
      ],
    ]);
    const boxPolygon = bboxPolygon(bbox(line));

    setDrawEntities([
      {
        coordinates: polygon.primitive,
        name: `${DrawType.BOX.toString()}_${timeStamp}`,
        id: timeStamp,
        type: DrawType.BOX,
        geojson: polygon.geojson,
      },
    ]);

    props.handlePolygonSelected((boxPolygon as Feature).geometry as Polygon); 
  };

  const onReset = (): void => {
    setDrawEntities([]);
    props.handlePolygonReset();
  };

  const onCancelDraw = (): void => {
    setIsDrawing(false);
    setDrawPrimitive(noDrawing);
  };

  const generateSelectedLayers = ():JSX.Element[] => {
    const arr = MOCK_DATA_IMAGERY_LAYERS_ISRAEL.map((layer)=>{
      return <CesiumXYZLayer key={layer.id} options={{url: layer.properties.url}}/>
    });

    return arr;
  };

  return (
    <div className="map">
      <div className="filtersPosition" style={{backgroundColor: theme.primary, width: props.mapActionsWidth}}>
        <div className="filtersContainer">
          <PolygonSelectionUi
            onCancelDraw={onCancelDraw}
            onReset={onReset}
            onStartDraw={setDrawType}
            isSelectionEnabled={isDrawing}
            onPolygonUpdate={onPolygonSelection}
            mapActionsWidth = {props.mapActionsWidth}
            handleOtherDrawers={props.handleOtherDrawers}
          />
          {props.filters?.map((filter, index) => (
            <div key={index} className="filtersMargin">
              {filter}
            </div>
          ))}
        </div>
      </div>
      <CesiumMap 
        projection={CONFIG.MAP.PROJECTION}  
        center={center}
        zoom={CONFIG.MAP.ZOOM}
        sceneMode={CesiumSceneMode.SCENE2D}
        imageryProvider={false}
      >
        {props.mapContent}
        {generateSelectedLayers()}
        <CesiumDrawingsDataSource
          drawings={drawEntities}
          material={DRAWING_MATERIAL_COLOR}
          outlineColor={DRAWING_OUTLINE_COLOR}
          drawState={{
            drawing: isDrawing,
            type: drawPrimitive.type,
            handler: drawPrimitive.handler,
          }}
        />
      </CesiumMap>
    </div>
  );
};
