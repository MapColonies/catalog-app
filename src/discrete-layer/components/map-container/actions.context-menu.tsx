import { Box, IContextMenuData } from '@map-colonies/react-components';
import React from 'react';
import { ContextMenu } from './context-menu';

interface IActionsContextMenuProps extends IContextMenuData {}

export const ActionsContextMenu: React.FC<IActionsContextMenuProps> = ({ ...restProps }) => {
const { handleClose } = restProps;

  return (
    <ContextMenu
      menuTitle='Query Services'
      menuSections={[
       [
        <Box onClick={handleClose}>Buildings</Box>,
        <Box onClick={handleClose}>Roads</Box>,
        <Box onClick={handleClose}>Tiles Coordinates</Box>,
       ],
       [
        <Box onClick={handleClose}>Get Height</Box>
       ]
      ]}
      {...restProps}
    />
  );
};
