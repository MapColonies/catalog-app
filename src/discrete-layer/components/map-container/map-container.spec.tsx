import React from 'react';
import { shallow } from 'enzyme';
// eslint-disable-next-line
import '../../../__mocks__/confEnvShim';
import { MapContainer } from './map-container';

// Enzyme doesnâ€™t work properly with hooks in general, especially for `shallow` so this is the way to mock `react-intl` module.
// Enspired by https://github.com/formatjs/formatjs/issues/1477
jest.mock('react-intl', () => {
  /* eslint-disable */
  const reactIntl = require.requireActual('react-intl');
  const MESSAGES = require.requireActual('../../../common/i18n');
  const intl = reactIntl.createIntl({
    locale: 'en',
    messages: MESSAGES.default['en'],
  });

  return {
    ...reactIntl,
    useIntl: () => intl,
  };
  /* eslint-enable */
});

const polygonSelectedFn = jest.fn();
const polygonResetFn = jest.fn();

describe('MapContainer component', () => {
  afterEach(() => {
    polygonSelectedFn.mockClear();
    polygonResetFn.mockClear();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <MapContainer
        handlePolygonSelected={polygonSelectedFn}
        handlePolygonReset={polygonResetFn}
        mapContent={<span>test</span>}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('MapContent prop elements transfered to MapWrapper component', () => {
    const wrapper = shallow(
      <MapContainer
        handlePolygonSelected={polygonSelectedFn}
        handlePolygonReset={polygonResetFn}
        mapContent={<span>test</span>}
      />
    );

    expect(wrapper.children()).toHaveLength(2); // one is drawing layer, second is transfered
  });
});
