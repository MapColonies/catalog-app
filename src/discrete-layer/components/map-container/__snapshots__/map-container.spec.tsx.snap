// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer component renders correctly 1`] = `
<div
  className="map"
>
  <div
    className="filtersPosition"
    style={
      Object {
        "backgroundColor": "#24aee9",
        "width": "400px",
      }
    }
  >
    <div
      className="filtersContainer"
    >
      <PolygonSelectionUi
        handleOtherDrawers={[Function]}
        isSelectionEnabled={false}
        mapActionsWidth="400px"
        onCancelDraw={[Function]}
        onPolygonUpdate={[Function]}
        onReset={[Function]}
        onStartDraw={[Function]}
      />
      <div
        className="filtersMargin"
      >
        <ForwardRef
          icon="filter_alt"
          onClick={[Function]}
          outlined={true}
          theme={
            Array [
              "primaryBg",
              "onPrimary",
            ]
          }
        >
          <FormattedMessage
            id="filters.title"
          />
        </ForwardRef>
        <DrawerOpener
          isOpen={false}
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
  <fr
    center={
      Array [
        34.811,
        31.908,
      ]
    }
    imageryProvider={false}
    projection="EPSG:4326"
    sceneMode={2}
    zoom={14}
  >
    <span>
      test
    </span>
    <Memo(observerComponent) />
    <Memo(observerComponent) />
    <Memo(observerComponent) />
    <Pr
      drawState={
        Object {
          "drawing": false,
          "handler": [Function],
          "type": "UNKNOWN",
        }
      }
      drawingMaterial={
        Color {
          "alpha": 0.5,
          "blue": 0,
          "green": 1,
          "red": 1,
        }
      }
      drawings={
        Array [
          Object {
            "coordinates": Array [],
            "id": "",
            "name": "",
            "type": "UNKNOWN",
          },
        ]
      }
    />
  </fr>
</div>
`;
